samples/msgext-northwind-inventory-csharp/MessageExtensions/DiscountedSearchCommand.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Builder.Teams;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using Microsoft.Bot.Schema.Teams;
+using msgext_northwind_inventory_csharp.Models;
+using msgext_northwind_inventory_csharp.NorthwindDB;
+using msgext_northwind_inventory_csharp.Handlers;
+
+namespace msgext_northwind_inventory_csharp.MessageExtensions
+{
+    public static class DiscountedSearchCommand
+    {
+        public const string CommandId = "discountSearch";
+        private static int queryCount = 0;
+
+        public static async Task<MessagingExtensionResponse> HandleTeamsMessagingExtensionQueryAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, IConfiguration configuration, CancellationToken cancellationToken)
+        {
+            var categoryName = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "categoryName")?.Value as string);
```

Comment:
Can we handle if categoryName is null or blank?

Reply:
It would not be null, it meant to send a empty string